File: 2207/main.go
package main

import "fmt"

func main() {
	fmt.Println(maximumSubsequenceCount("abcdefghijklmnoplrsuvwz", "ac"))
}

func maximumSubsequenceCount(text string, pattern string) int64 {
	var res, cnt1, cnt2 int64
	for _, c := range text {
		if byte(c) == pattern[1] {
			res += cnt1
			cnt2++
		}
		if byte(c) == pattern[0] {
			cnt1++
		}
	}
	if cnt1 > cnt2 {
		return res + cnt1
	}
	return res + cnt2
}

File: 2207/main.rs
struct Solution {}

impl Solution {
    pub fn maximum_subsequence_count(text: String, pattern: String) -> i64 {
        let mut res = 0;
        let mut cnt1 = 0;
        let mut cnt2 = 0;
        for c in text.chars() {
            if c == pattern.chars().nth(1).unwrap() {
                res += cnt1;
                cnt2 += 1;
            }
            if c == pattern.chars().nth(0).unwrap() {
                cnt1 += 1;
            }
        }
        res + std::cmp::max(cnt1, cnt2)
    }
}

fn main() {
    println!("{}", Solution::maximum_subsequence_count("abcdefj".to_string(), "ac".to_string()));
}